### PR TITLE
topology: Update sof-tgl-max98373-rt5682 to take amp SSP ID

### PR DESCRIPTION
**Pull request by sathya-nujella**
Update sof-tgl-max98373-rt5682 to take speaker amp's SSP ID as input.
This is to accommodate DUT's which has amp connected on SSP2.

### CODE REVIEW

File: tools/topology/sof-tgl-max98373-rt5682.m4 [hunk 1]
```diff
@@ -37,12 +37,21 @@ DEBUG_START
...
+define(`SMART_SSP_INDEX', AMP_SSP)
 #define SSP BE dai_link name
+ifelse(AMP_SSP, `1',
+`
 define(`SMART_SSP_NAME', `SSP1-Codec')
+', `')
+ifelse(AMP_SSP, `2',
+`
+define(`SMART_SSP_NAME', `SSP2-Codec')
+', `')
```
Comment (keyonjie) [inline tools/topology/sof-tgl-max98373-rt5682.m4:54]: We can use concat then no ifelse needed, something like:
define(`SMART_SSP_NAME', concat(concat(`SSP', AMP_SSP), `-Codec'))

Comment (sathya-nujella) [reply tools/topology/sof-tgl-max98373-rt5682.m4:54]: oh string concatenation is there. Thank you @keyonjie .
Let me try and update.

Comment (sathya-nujella) [reply tools/topology/sof-tgl-max98373-rt5682.m4]: > We can use concat then no ifelse needed, something like:
> 
> ```
> define(`SMART_SSP_NAME', concat(concat(`SSP', AMP_SSP), `-Codec'))
> ```

Done, updated.

**Comment by sathya-nujella** [comment]
Build error here, seems not related to this PR:
sof-ci/jenkins/pr-device-test — On-device Test Failed:BYT_MB_RT5682

**Comment by jajanusz** [comment]
@bkokoszx you made smartams, please check this

**Comment by sathya-nujella** [comment]
> @bkokoszx you made smartams, please check this

For ref:
We have verified on both of the below DUT's:
a. Volteer with max98373 on I2S1
b. DUT with Max98373 on I2S2
On both of them, speaker playback works.

**Comment by sathya-nujella** [comment]
> Build error here, seems not related to this PR:
> sof-ci/jenkins/pr-device-test — On-device Test Failed:BYT_MB_RT5682

@keyonjie , @xiulipan 
Can you please review and suggest ?  I don't see why BYT fails with this.

**Comment by plbossart** [comment]
> > Build error here, seems not related to this PR:
> > sof-ci/jenkins/pr-device-test — On-device Test Failed:BYT_MB_RT5682
> 
> @keyonjie , @xiulipan
> Can you please review and suggest ? I don't see why BYT fails with this.

@sathya-nujella ignore this error, it's a known problem already tracked.

**Pull request merged**